### PR TITLE
FAD-6040: AWS Marketplace experience

### PR DESCRIPTION
**Pull request by rajumsys**
Changes:
- [x] Signup page logo 
- [x] Billing page experience
- [x] Tests for new selectors
- [x] IP purchase ux

Remaining:
- [x] Onboarding. AWS users should be to pick one of the valid (aws only) plans. Shouldn't need to enter card info.

Doc: https://confluence.int.messagesystems.com/display/ENG/AWS+Marketplace#AWSMarketplace-Screenshots

Note:

### CODE REVIEW

File: src/pages/billing/SummaryPage.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ import { Page, Panel, WindowEvent } from '@sparkpost/matchbox';
```
Comment (jasonrhodes) [inline src/pages/billing/SummaryPage.js]: This is the file we'll have conflicts in, we can talk and make a plan depending on who is going to merge first.

File: src/actions/billing.js [hunk 2]
```diff
@@ -19,12 +19,12 @@ export function syncSubscription() {
  * Updates plan
  * @param {string} code
  */
-export function updateSubscription(code) {
+export function updateSubscription(code, isAWSAccount = false) {
```
[the remark this answers is not in the document]
Comment (bizob2828) [reply src/actions/billing.js]: i don't see tests for when isAWSAccount is true

File: src/components/planPicker/Plan.js [hunk 3]
```diff
@@ -1,32 +1,53 @@
...
+      <a className={className} {...rest} >
+        <span
+          className={styles.MainLabel}><strong>{ plan.volume.toLocaleString() }</strong> emails/month for {this.getPrice(plan)}</span>
+        <span className={styles.SupportLabel}>{ overage }{ ip }</span>
+      </a>
+    );
+  }
+}
+
+function eitherMonthlyOrHourly(props, propName, componentName) {
```
Comment (bizob2828) [inline src/components/planPicker/Plan.js]: not sure if you can test but don't see tests for it

File: src/pages/billing/components/Confirmation.js [hunk 4]
```diff
@@ -1,87 +1,110 @@
...
-          <p>Note: your current plan includes a free dedicated IP address.</p>
-          <p>If you downgrade to the selected plan, you will lose that discount and will be charged the standard ${ config.sendingIps.pricePerIp } / month price for each dedicated IP on your next statement.</p>
-          <p>To remove dedicated IPs from your account, please <UnstyledLink to={`mailto:${config.contact.supportEmail}`}>contact our support team</UnstyledLink>.</p>
-        </div>
-      );
-    }
+    if (isPlanSelected && billingEnabled) {
+      if (!isDowngrade) {
+        effectiveDateMarkup = current.isFree
+          ? <p>Your upgrade will be effective today.</p>
```
Comment (bizob2828) [inline src/pages/billing/components/Confirmation.js]: missing a test for this branch

File: src/pages/billing/forms/ChangePlan.js [hunk 5]
```diff
@@ -91,32 +91,28 @@ export class ChangePlan extends Component {
   }
 
   render() {
-    const { account, submitting, pristine, currentPlan, selectedPlan, plans } = this.props;
+    const { account, submitting, pristine, currentPlan, selectedPlan, plans, isAWSAccount } = this.props;
+    const billingEnabled = account.subscription.self_serve || isAWSAccount;
```
Comment (bizob2828) [inline src/pages/billing/forms/ChangePlan.js]: missing a test where isAWSAccount is true

File: src/config/default.js [hunk 6]
```diff
@@ -54,7 +54,8 @@ const config = {
   },
   sendingIps: {
     maxPerAccount: 4,
-    pricePerIp: 20.00
+    pricePerIp: 20.00,
+    awsPricePerIp: 0.028
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/config/default.js]: treasure hunt https://repos.int.messagesystems.com/hg/MessageSystems/Ansible-Deployment/file/1a8f932a3ffc/roles/ui_tier/templates/overrides.js.j2#l59

File: src/actions/billing.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/actions/billing.js]: right, missed that. i'll add one.

File: src/components/planPicker/Plan.js (hunk 3 above)
Comment (rajumsys) [reply src/components/planPicker/Plan.js]: to test this method, i'll have to export it which is unnecessary as I've tested the component with different attr (hourly, monthly).  i think another way of testing will be passing both or neither and assert on warning :).

File: src/pages/billing/forms/ChangePlan.js (hunk 5 above)
Comment (rajumsys) [reply src/pages/billing/forms/ChangePlan.js]: test added but as you commented in the source file, it's not marked as outdated!

File: src/pages/billing/components/SummarySection.js [hunk 7]
```diff
@@ -1,31 +1,21 @@
...
     return <LabelledValue label='Your Plan' />;
   }
 
-  if (plan.monthly !== undefined) {
-    monthly = plan.monthly === 0
-      ? 'for Free'
-      : <span><strong>for ${plan.monthly.toLocaleString()}</strong> per month</span>;
-  }
-
   const overage = plan.overage
```
Comment (bizob2828) [inline src/pages/billing/components/SummarySection.js]: can't you use the showOverage param here?

**Pull request merged**